src/builtins/core/duration/date.rs

```diff
@@ -105,4 +108,60 @@ impl DateDuration {
     pub fn sign(&self) -> Sign {
         super::duration_sign(&self.fields())
     }
+
+    /// DateDurationDays
+    /// TODO(review): Question: what the correct return type?
+    pub fn days(&self, relative_to: &PlainDate) -> TemporalResult<i64> {
+        // 1. Let yearsMonthsWeeksDuration be ! AdjustDateDurationRecord(dateDuration, 0).
+        let ymw_duration = self.adjust(FiniteF64(0.0), None, None)?;
+        // 2. If DateDurationSign(yearsMonthsWeeksDuration) = 0, return dateDuration.[[Days]].
+        if ymw_duration.sign() == Sign::Zero {
+            return self.days.as_integer_if_integral();
+        }
+        // 3. Let later be ? CalendarDateAdd(plainRelativeTo.[[Calendar]], plainRelativeTo.[[ISODate]], yearsMonthsWeeksDuration, constrain).
+        let later = relative_to.add(
+            &Duration {
+                date: *self,
+                time: Default::default(),
+            },
+            Some(ArithmeticOverflow::Constrain),
+        )?;
+        // 4. Let epochDays1 be ISODateToEpochDays(plainRelativeTo.[[ISODate]].[[Year]], plainRelativeTo.[[ISODate]].[[Month]] - 1, plainRelativeTo.[[ISODate]].[[Day]]).
+        let epoch_days_1 = iso_date_to_epoch_days(
+            relative_to.iso_year(),
+            i32::from(relative_to.iso_month()) - 1,
+            i32::from(relative_to.iso_day()),
+        );
+        // 5. Let epochDays2 be ISODateToEpochDays(later.[[Year]], later.[[Month]] - 1, later.[[Day]]).
+        let epoch_days_2 = iso_date_to_epoch_days(
+            later.iso_year(),
+            i32::from(later.iso_month()) - 1,
+            i32::from(later.iso_day()),
+        );
+        // 6. Let yearsMonthsWeeksInDays be epochDays2 - epochDays1.
+        let ymd_in_days = epoch_days_2 - epoch_days_1;
+        // 7. Return dateDuration.[[Days]] + yearsMonthsWeeksInDays.
+        Ok(self.days.as_integer_if_integral::<i64>()?
+            + i64::try_from(ymd_in_days).map_err(|_| TemporalError::range())?)
+    }
+
+    /// AdjustDateDurationRecord
+    pub fn adjust(
```

Comment:
suggestion: `pub(crate)`

Since this is an abstract op, it would be best to keep these `pub(crate)` until otherwise needed.